mathics/eval/arithmetic.py

```diff
@@ -0,0 +1,812 @@
+            or isinstance(exp, Rational)
+            and exp.value.p < 0
+        ):
+            base, exp = eval_inverse_number(base), eval_negate_number(exp)
+            return eval_power_number(base, exp) or Expression(SymbolPower, base, exp)
+
+        p, q = (Integer(u) for u in base.value.as_numer_denom())
+        p_eval, q_eval = (eval_power_number(u, exp) for u in (p, q))
+        # If neither p^exp or q^exp produced a new result,
+        # leave it alone
+        if q_eval is None and p_eval is None:
+            return None
+        # if q^exp == 1: return p_eval
+        # (should not happen)
+        if q_eval is Integer1:
+            return p_eval
+        if isinstance(q_eval, Integer):
+            if isinstance(p_eval, Integer):
+                return Rational(p_eval.value, q_eval.value)
+
+        if p_eval is None:
+            p_eval = Expression(SymbolPower, p, exp)
+
+        if q_eval is None:
+            q_eval = Expression(SymbolPower, q, exp)
+        return Expression(
+            SymbolTimes, p_eval, Expression(SymbolPower, q_eval, IntegerM1)
+        )
+    # Pure imaginary base case
+    elif isinstance(base, Complex) and base.real.is_zero:
+        base = base.imag
+        if base.value < 0:
+            base = eval_negate_number(base)
+            phase = Expression(
+                SymbolPower,
+                IntegerM1,
+                eval_multiply_numbers([IntegerM1, RationalOneHalf, exp]),
+            )
+        else:
+            phase = Expression(
+                SymbolPower, IntegerM1, eval_multiply_numbers([RationalOneHalf, exp])
+            )
+        real_factor = eval_power_number(base, exp)
+
+        if real_factor is None:
+            return None
+        return Expression(SymbolTimes, real_factor, phase)
+
+    # Generic case
+    return eval_power_sympy()
+
+
+def eval_power_inexact(base: Number, exp: Number) -> BaseElement:
+    """
+    Eval base^exp for `base` and `exp` inexact numbers
+    """
+    # If both base and exponent are exact quantities,
+    # use sympy.
+    prec = min_prec(base, exp)
+    if prec is not None:
+        is_machine_precision = base.is_machine_precision() or exp.is_machine_precision()
+        if is_machine_precision:
+            number = mpmath.power(base.to_mpmath(), exp.to_mpmath())
+            return from_mpmath(number)
+        else:
+            with mpmath.workprec(prec):
+                number = mpmath.power(base.to_mpmath(), exp.to_mpmath())
+                return from_mpmath(number, dps(prec))
+
+
+def eval_sign(expr: BaseElement) -> Optional[BaseElement]:
+    """
+    evaluates Sign[expr]
+    """
+    if isinstance(expr, Atom):
+        return eval_sign_number(expr)
+    if expr.has_form("Power", 2):
+        base, exp = expr.elements
+        if exp.is_zero:
+            return Integer1
+        if isinstance(exp, (Integer, Real, Rational)):
+            sign = eval_sign(base) or Expression(SymbolSign, base)
```

Comment:
My Editor tells and flake8 tell me SymbolSign is not defined.

Reply:
Thanks for catching this.

Reply:
I have to add more tests for these routines, but at this point, I need a little bit of feedback (not necessarily right now, but before continuing with this).